src/ruby/puppet-server-lib/puppet/server/oids.rb

```diff
@@ -0,0 +1,31 @@
+module Puppet::Server::Oids
+  def self.load_custom_oid_file(custom_oid_file, map_key='oid_mapping')
+    if File.exists?(custom_oid_file) && File.readable?(custom_oid_file)
```

Comment:
From what I can tell, the `readable?` check would be sufficient on its own. However, this function was copied straight from the Puppet codebase with minor modifications (it returns `oid_defns` instead of iterating through and registering everything in it with OpenSSL), so I wasn't sure if we should keep parity with the way that function actually performs the file parsing